### ISSUE TITLE
How BERT Works

### ISSUE THREAD

**Issue by mderosa001** [mderosa001/Capstone_Issues#3]
### Summary 
BERT (Bidirectional Encoder Representations from Transformers) is an open-source machine learning framework for natural language processing (NLP). It helps computers understand ambiguous language by considering the surrounding text for context. BERT was pretrained on Wikipedia and can be fine-tuned for specific tasks like question answering. **How BERT Works** Bidirectionality: Unlike previous models that processed text either left-to-right or right-to-left, BERT reads in both directions simultaneously.Transformer Model: Uses deep learning where all input elements are connected dynamically. Masked Language Modeling (MLM): Hides a word in a sentence and predicts it using context. Next Sentence Prediction (NSP): Determines whether two sentences are logically connected. Background and Impact Introduced by Google in 2018, based on Transformer models (originally developed in 2017). Outperformed previous models like word2vec and GloVe, which struggled with polysemous (multiple-meaning) words. Applied to Google Search in 2019, improving 10% of U.S. English queries and later expanding to 70+ languages. Inspired models like GPT-2 and ChatGPT.

### Key Points 
Self-Attention Mechanisms: Analyzes relationships between words for better meaning extraction. Transfer Learning: Pretrained on vast text data and fine-tuned for specific NLP tasks. Open Source: Developers can train models on cloud TPUs or GPUs.

### Citation 
@misc{Hashemi-Pour_Lutkevich_2024, title={What is the bert language model?: Definition from TechTarget}, url={https://www.techtarget.com/searchenterpriseai/definition/BERT-language-model}, journal={Search Enterprise AI}, publisher={TechTarget}, author={Hashemi-Pour, Cameron and Lutkevich, Ben}, year={2024}, month={Feb}}
